pasta/base/codegen.py

```diff
@@ -62,6 +63,33 @@ def visit_Str(self, node):
     self.code += content if content is not None else repr(node.s)
     self.suffix(node)
 
+  def visit_JoinedStr(self, node):
+    self.prefix(node)
+    content = fmt.get(node, 'content')
+
+    if content is None:
+      parts = []
+      for val in node.values:
+        if isinstance(v, ast.Str):
+          parts.append(v.s)
+        else:
+          parts.append('{__pasta_fstring_val_%d__}' % len(parts))
```

Comment:
seems like this should be a constant somewhere since it shows up in a couple places across diferent files like maybe an fstring_utils.py?

Reply:
Probably should, yes! Done. Extracted some other fstring logic there too.